src/types.rs

```diff
@@ -15,6 +15,14 @@ pub struct VersionData {
     pub version: String,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Coordinate {
+    #[serde(with = "crate::serde::as_string")]
+    slot: Slot,
+    root: Root,
+    execution_optimistic: bool,
```

Comment:
if we keep this we will prolly want `pub` on all of these

Reply:
```suggestion
    pub slot: Slot,
    pub root: Root,
    pub execution_optimistic: bool,
```